addons/hr_timesheet/models/project_project.py

```diff
@@ -138,26 +142,43 @@ def _compute_total_timesheet_time(self):
             total_time *= project.timesheet_encode_uom_id.factor
             project.total_timesheet_time = int(round(total_time))
 
+    def copy(self, default=None):
+        new_projects = super().copy(default=default)
+        analytic_fnames = self._get_plan_fnames()[1:]
+        for old_project, new_project in zip(self, new_projects):
+            if new_project.allow_timesheets:
+                new_project.write({
+                    fname: old_project[fname].id
+                    for fname in analytic_fnames
+                })
+        return new_projects
+
     @api.model_create_multi
     def create(self, vals_list):
         """ Create an analytic account if project allow timesheet and don't provide one
             Note: create it before calling super() to avoid raising the ValidationError from _check_allow_timesheet
         """
-        defaults = self.default_get(['allow_timesheets', 'analytic_account_id'])
-        analytic_accounts_vals = [vals for vals in vals_list if
-            vals.get('allow_timesheets', defaults.get('allow_timesheets'))
-            and not vals.get('analytic_account_id', defaults.get('analytic_account_id'))
+        defaults = self.default_get(['allow_timesheets', 'account_id'])
+        analytic_accounts_vals = [
+            vals for vals in vals_list
+            if (
+                vals.get('allow_timesheets', defaults.get('allow_timesheets')) and
+                not vals.get('account_id', defaults.get('account_id'))
+            )
```

Comment:
It can be a precomputed field if you want to avoid an update.
Why don't you think it's a good idea?

Reply:
Because it would mean we will generate an analytic account each time the `account_id` field is empty and `allow_timesheets` field is True and so if the user changes the account_id, the compute could generate each time a new analytic account and so we will have many useless analytic account.